server/utils/nunjucksSetup.test.ts

```diff
@@ -1,190 +1,90 @@
-import * as cheerio from 'cheerio'
-import nunjucks, { Template } from 'nunjucks'
-import { registerNunjucks } from './nunjucksSetup'
+import express from 'express'
+import { FieldValidationError } from 'express-validator'
+import nunjucksSetup from './nunjucksSetup'
 
 describe('Nunjucks Filters', () => {
-  let compiledTemplate: Template
-  let viewContext: Record<string, unknown>
-
-  const njkEnv = registerNunjucks(null)
-
-  describe('initialiseName', () => {
-    it('should return null if full name is not provided', () => {
-      viewContext = {}
-      const nunjucksString = '{{ fullName | initialiseName }}'
-      compiledTemplate = nunjucks.compile(nunjucksString, njkEnv)
-      const $ = cheerio.load(compiledTemplate.render(viewContext))
-      expect($('body').text()).toBe('')
-    })
-
-    it('should return formatted name', () => {
-      viewContext = {
-        fullName: 'Joe Bloggs',
-      }
-      const nunjucksString = '{{ fullName | initialiseName }}'
-      compiledTemplate = nunjucks.compile(nunjucksString, njkEnv)
-      const $ = cheerio.load(compiledTemplate.render(viewContext))
-      expect($('body').text()).toBe('J. Bloggs')
-    })
-  })
+  const app = express()
+  const njk = nunjucksSetup(app, null)
 
   describe('errorSummaryList', () => {
     it('should map errors to text and href', () => {
-      viewContext = {
-        errors: [
-          {
-            msg: 'Field 1 message',
-            path: 'field1',
-          },
-          {
-            msg: 'Field 2 message',
-            path: 'field2',
-          },
-        ],
-      }
-      const nunjucksString = '{{ errors | errorSummaryList | dump }}'
-      compiledTemplate = nunjucks.compile(nunjucksString, njkEnv)
-      const $ = cheerio.load(compiledTemplate.render(viewContext))
-      expect($('body').text()).toBe(
-        JSON.stringify([
-          {
-            text: 'Field 1 message',
-            href: '#field1-error',
-          },
-          {
-            text: 'Field 2 message',
-            href: '#field2-error',
-          },
-        ]),
-      )
+      const errors = <FieldValidationError[]>[
+        { msg: 'Field 1 message', path: 'field1' },
+        { msg: 'Field 2 message', path: 'field2' },
+      ]
+      const expectedResult = [
+        { text: 'Field 1 message', href: '#field1-error' },
+        { text: 'Field 2 message', href: '#field2-error' },
+      ]
+
+      const result = njk.getFilter('errorSummaryList')(errors)
```

Comment:
Changed to a much simpler way of testing Nunjucks filters by just calling the filter functions directly instead of loading them into a template and compiling it.